pkg/apis/v1alpha1/vpc_types.go

```diff
@@ -0,0 +1,138 @@
+	   // Service Gateway serves as the default gateway for VPC.
+	   // +optional
+	   ServiceGateway ServiceGatewayItem `json:"serviceGateway"`
+	   // This is policy resource path of pre-created public ip block which is created by org-admin at
+	   // org level under org-default-project.
+	   // +optional
+	   PublicIPv4Block string `json:"publicIPv4Block"`
+	   // IP block for private CIDR used for network only within VPC project, resource path should be
+	   // located under /orgs/default/projects/project-1/infra/ip-blocks/private-block, this is optional.
+	   // +optional
+	   PrivateCIDRs string `json:"privateCIDRs"`
+	   // DHCP config specified at VPC level is applicable to all the subnets under the VPC.
+	   // +optional
+	   DHCPConfig DHCPConfigItem `json:"dhcpConfig"`
+   }
+   
+   type ServiceGatewayItem struct {
+	   // Flag to disable service gateway for connected subnets.
+	   // If false then VPC will support the following services: NAT, gateway security policies, and gateway QoS profile.
+	   // If true, VPC will support only distributed services i.e., EW security policy. Workload shall be protected using the same.
+	   // All traffic from subnets will be routed through the distributed router to the connected gateway and eliminates the intermediate hop of the service gateway.
+	   // +optional
+	   // +kubebuilder:default=false
+	   Disable bool `json:"disable"`
+	   // Auto plumb snat rule for private subnet, this will make sure private subnets are routable outside of VPC.
+	   // There will be one snat rule per VPC, translated ip will be taken from public ip block. If enabled, user must also
+	   // configure public ip block. This property is applicable only when service_gateway is enabled.
+	   // +optional
+	   // +kubebuilder:default=true
+	   AutoSNAT string `json:"autoSnat"`
+   }
+   
+   type DHCPConfigItem struct {
+	   // If enabled, the DHCP server will be configured based on IP address type.
+	   // If disabled then neither DHCP server nor relay shall be configured.
+	   // +optional
+	   // +kubebuilder:default=true
+	   EnableDHCPServer bool `json:"enable_dhcp_server"`
+	   // Policy path of DHCP-relay-config. If configured then all the subnets will be configured with the DHCP relay server.
+	   // If not specified, then the local DHCP server will be configured for all connected subnets.
+	   // +optional
+	   DHCPRelayConfigPath string `json:"dhcp_relay_config_path"`
+	   // IPs in % to be reserved for DHCP ranges. By default, 80% of IPv4 IPs will be reserved for DHCP
+	   // ranges applied on the subnet. Configure 0 if no pool is required.
+	   // +optional
+	   // +kubebuilder:default=80
+	   DHCPIPv4PoolPercentage int `json:"dhcp_ipv4_pool_percentage"`
+	   //Dns configuration for VPC.
+	   // +optional
+	   DNSConfig DNSConfigItem `json:"dnsConfig"`
+   }
+   
+   type DNSConfigItem struct {
+	   // IP address list of DNS servers.
+	   // +optional
+	   DNSIPs []string `json:"dnsIPs"`
```

Comment:
Name it DNSServers?

Reply:
How about DNSServerIPs ? this is the dhcp_config in NSX-T VPC API.
 "dhcp_config": {
            "enable_dhcp": true,
            "dhcp_v4_pool": 50,
            "dhcp_v6_pool": 500,
            "dns_client_config": {
                "dns_server_ips": [
                    "value: \"192.168.1.25\"\n"
                ]
            }
        },

Reply:
Sure, we should follow NSX-T API definitions when that makes sense.